### PR TITLE
fix shape/dtype/storage inference.

### PR DESCRIPTION
**Pull request by zheng-da**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/subgraph/subgraph_op.cc [hunk 1]
```diff
@@ -154,10 +154,27 @@ bool SubgraphOpShape(const nnvm::NodeAttrs& attrs,
   CHECK_EQ(idx_g.input_nodes().size(), in_shapes->size());
   CHECK_EQ(idx_g.outputs().size(), out_shapes->size());
   // TODO: make sure shape inputs matches the order from in_shapes
-  nnvm::ShapeVector shape_inputs = *in_shapes;
-  imperative::CheckAndInferShape(&g, std::move(shape_inputs), true);
-  const nnvm::ShapeVector& shapes = g.GetAttr<nnvm::ShapeVector>("shape");
-  const std::vector<uint32_t>& input_nids = idx_g.input_nodes();
+
+  // Put the input and output shapes to the shape vector.
+  nnvm::ShapeVector shapes(idx_g.num_nodes());
```
Comment (reminisce) [inline src/operator/subgraph/subgraph_op.cc]: You are populating known shapes with entry ids. I think this one should be allocated with size equal to `idx_g.num_node_entries()` and attache attribute `shape` to the graph.
